### PR TITLE
Uploadtooll

### PR DESCRIPTION
**Pull request by Adrianzatreanu**

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
Hi @justuswilhelm this is the new PR i created due to those git issues, so feel free to ack it whenever you have time :) Thanks

**Comment by sils1297** [comment]
@Adrianzatreanu didn't you want to reack?

File: bears/generate_package.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    os.makedirs(upload_package_folder, exist_ok=True)
+    touch(os.path.join(upload_package_folder, '__init__.py'))
+    shutil.copyfile(file_to_copy, os.path.join(upload_package_folder,
+                                               object_name + '.py'))
+
+
+def perform_upload(path):
+    """
+    Uploads the directory to PyPi, creating a ``sdist`` and a ``bdist_wheel``,
+    after registering it.
```
Comment (sils1297) [inline bears/generate_package.py]: you don't always want to register it, only at the first time! Registering should be a flag like `coala-bears --register --upload` would register *and* upload it

File: bears/generate_package.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    """
+    Creates a parser for command line arguments.
+
+    :return: Parser arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description='Generates PyPi packages from bears.')
+    parser.add_argument('-u', '--upload', help='Upload the packages on PyPi',
+                        action='store_true')
+    return parser
```
Comment (sils1297) [inline bears/generate_package.py]: I'd return `parser.parse_args()` and adjust docs accordingly

File: bears/generate_package.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+
+def main():
+    args = create_upload_parser().parse_args()
+
+    os.makedirs(os.path.join('bears', 'upload'), exist_ok=True)
+
+    bear_file_name_list = glob.glob('bears/*/*Bear.py') + (
+        (glob.glob('bears/*/*/*Bear.py')))
```
Comment (sils1297) [inline bears/generate_package.py]: c'me on, it could always be deeper, you need a recursive glob

Comment (sils1297) [inline bears/generate_package.py]: just use `bears/**/*Bear.py` with the coala glob

File: bears/generate_package.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    for bear_file_name in bear_file_name_list:
+        bear_object = next(iimport_objects(
+            bear_file_name, attributes='kind', local=True),
+            None)
```
Comment (sils1297) [inline bears/generate_package.py]: why are you taking only the first? What's with the others?

File: bears/generate_package.py (hunk 2 above)
Comment (Adrianzatreanu) [inline bears/generate_package.py]: the way i made it like this was to be easier used and tested, i failed to test it with the return you mention it, gave a lot of errors that said nothing concrete. so id rather leave it like this if its okay.

File: bears/generate_package.py (hunk 4 above)
Comment (Adrianzatreanu) [inline bears/generate_package.py]: what others? is there the slightest chance of having more bears in the same file?

**Comment by Adrianzatreanu** [comment]
reack 763aa4e

File: bears/generate_package.py (hunk 4 above)
Comment (sils1297) [inline bears/generate_package.py]: yes! We even have a mechanism built into coala for it.

Comment (Adrianzatreanu) [inline bears/generate_package.py]: so how am i treating those..? :/

**Comment by justuswilhelm** [comment]
ack f7836d1

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Pull request merged**